ui/src/shared/components/TimeMachine/TimeMachine.tsx

```diff
@@ -139,6 +139,7 @@ export const FluxContext = React.createContext(undefined)
 
 class TimeMachine extends PureComponent<Props, State> {
   private debouncedASTResponse: ScriptFunc
+  private validAST: boolean = true
```

Comment:
is this not stored in the component state to avoid rerenders?

Reply:
yeah